MosipNexus/api/main.py

```
@@ -0,0 +1,529 @@
+@app.post("/batch", response_model=BatchResponse, tags=["Chat"],
+          summary="Ask multiple questions at once",
+          description="Process up to 10 questions sequentially in a single request. Each question is aware of the previous answers (shared session context).")
+def batch(req: BatchRequest) -> BatchResponse:
+    session_id = req.session_id or str(uuid.uuid4())
+    memory = _get_session(session_id)
+    results: list[ChatResponse] = []
+
+    for question in req.questions:
+        try:
+            result = ask(question, memory.messages, req.language)
+        except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)